### PR TITLE
Favorite Icon Action In ParkingDetailScreen

### LINKED ISSUE #161: As a user I want to save my favorite spots, so that I can find them easily.

### PR DESCRIPTION
**Pull request by AlexandreBeeDeeBop**
Content of this PR : 
This small PR adds the possibility for a user to add a parking to his list of favorite spots via the ParkingDetailScreen.

How and why : 
This is the first way to add a parking spot to the favorites, the second one will hopefully come with my next PR in the list screen.
By clicking on the heart icon on the top right of the screen, the user can add or remove the parking from his preferences.
When the heart is empty and outlined in black, it means the parking is not yet in the favorites.
On the other hand, when the heart is filled in red, it means the parking is already in the favorites.
Obviously, the user can add or remove that parking from his favorites.
A user who is not yet signed in will still see the empty black heart but wont be able to use it, instead that guest user will get a message inviting him to register.

Test Report : 
image

Closes #161

### CODE REVIEW

**Review by Armetiis — APPROVED** [review]
Good job. Works as intended, I have nothing to add except the two little remarks I left you.

This is a bit unrelated but maybe you could try to take this chance to change the star into a heart in the Profile screen? Nonblocking, you have my approval

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt [hunk 1]
```diff
@@ -70,8 +76,51 @@ fun ParkingDetailsScreen(
...
+                                          selectedParking.uid)
+                                    } else {
+                                      userViewModel.addFavoriteParkingToSelectedUser(
+                                          selectedParking.uid)
+                                    }
+                                    userViewModel.getSelectedUserFavoriteParking()
+                                  } else {
+                                    Toast.makeText(
+                                            context,
+                                            "Please sign in to add favorites",
```
[the remark this answers is not in the document]
Comment (AlexandreBeeDeeBop) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:117]: None of the other toasts have used string resource. And the string resources do not exist. Do i create them ?

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt [hunk 2]
```diff
@@ -70,8 +76,51 @@ fun ParkingDetailsScreen(
...
+                  contentAlignment = Alignment.TopEnd) {
+                    val isFavorite =
+                        if (userSignedIn.value) {
+                          userViewModel.favoriteParkings
+                              .collectAsState()
+                              .value
+                              .contains(selectedParking)
+                        } else {
+                          false
+                        }
```
[the remark this answers is not in the document]
Comment (AlexandreBeeDeeBop) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt]: I guess...

File: app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/parkingDetails/ParkingDetailsScreen.kt:114]: Yes you have to create it. I think ones that do not have a string resource are temporary (as opposed to yours). There was a comment to say they are temporary (to justify the absence of need of resources) but you removed them 😂

**Review by Armetiis — APPROVED** [review]

**Review by LouisJamesVasseur — APPROVED** [review]

**Pull request merged**